rbc/tests/heavydb/test_text.py

```diff
@@ -152,3 +152,23 @@ def ne2(a):
     assert ne1('hello', 'hello2').execute() == 1
 
     assert ne2('world').execute() == 0
+
+
+def test_if_else_assignment(heavydb):
+    heavydb.require_version((5, 8), 'Requires heavydb 5.8 or newer')
+
+    # fix for issue #487
+
+    @heavydb("TextEncodingNone(double)")
+    def classify_slope(slope):
+        if slope <= 5:
+            res = TextEncodingNone("low")
+        elif 5 < slope < 15:
+            res = TextEncodingNone("med")
+        else:
+            res = TextEncodingNone("high")
+        return res
```

Comment:
Would it be possible to support the following
```
    @heavydb("TextEncodingNone(double)")
    def classify_slope(slope):
        if slope <= 5:
            res = "low"
        elif 5 < slope < 15:
            res = "med"
        else:
            res = "high"
        return TextEncodingNone(res)
```
or even:
```
    @heavydb("TextEncodingNone(double)")
    def classify_slope(slope):
        if slope <= 5:
            res = "low"
        elif 5 < slope < 15:
            res = "med"
        else:
            res = "high"
        return res
```
? So that users would not need to use `TextEncodingNone` inside the Python function.

Reply:
As far as I can tell, unicode support requires the Numba runtime. For simple cases like the one above, we can cast the `StringLiteral` to `TextEncodingNone`. But anything other than that is not possible out of the box without NRT enabled. For instance, the code below fails with `NRT required but not enabled` error.

```python
@heavydb("TextEncodingNone(int32)")
def foo(x):
    s = str(x)
    return s

print(foo(3).execute())
```

```
E   No implementation of function Function(<class 'str'>) found for signature:
E    
E    >>> str(int32)
E    
E   There are 6 candidate implementations:
E       - Of which 4 did not match due to:
E       Overload of function 'str': File: <numerous>: Line N/A.
E         With argument(s): '(int32)':
E        No match.
E       - Of which 2 did not match due to:
E       Overload in function 'integer_str': File: numba/cpython/unicode.py: Line 2394.
E         With argument(s): '(int32)':
E        Rejected as the implementation raised a specific error:
E          NumbaRuntimeError: Failed in nopython mode pipeline (step: native lowering)
E        NRT required but not enabled
E        During: lowering "s = call $76load_global.17(kind, char_width, length, $const84.21, func=$76load_global.17, args=[Var(kind, unicode.py:2408), Var(char_width, unicode.py:2409), Var(length, unicode.py:2407), Var($const84.21, unicode.py:2410)], kws=(), vararg=None, target=None)" at /home/guilhermeleobas/miniconda3/envs/rbc/lib/python3.10/site-packages/numba/cpython/unicode.py (2410)
E     raised from /home/guilhermeleobas/miniconda3/envs/rbc/lib/python3.10/site-packages/numba/core/runtime/context.py:17
E   
E   During: resolving callee type: Function(<class 'str'>)
E   During: typing of call at /home/guilhermeleobas/git/rbc/rbc/tests/heavydb/test_text.py (161)
E   
E   
E   File "rbc/tests/heavydb/test_text.py", line 161:
E       def foo(x):
E           s = str(x)
E           ^
```

Reply:
OK. Let me provide another example:
```python
    @heavydb("TextEncodingNone(double)")
    def classify_slope(slope):
        if slope <= 5:
            res = b"low"
        elif 5 < slope < 15:
            res = b"med"
        else:
            res = b"high"
        return res
```
Would using bytes also require NRT?

Reply:
Yes, bytes also require NRT.

Reply:
OK, thanks! Let's land this PR and address the NRT problem in a possible follow-up.